library/core/src/iter/adapters/zip.rs

```diff
@@ -254,10 +285,12 @@ where
 }
 
 #[doc(hidden)]
-impl<A, B> ZipImpl<A, B> for Zip<A, B>
+impl<A, B> const ZipImpl<A, B> for Zip<A, B>
 where
-    A: TrustedRandomAccess + Iterator,
-    B: TrustedRandomAccess + Iterator,
+    A: ~const TrustedRandomAccess + ~const Iterator,
+    B: ~const TrustedRandomAccess + ~const Iterator,
+    A::Item: ~const Destruct,
+    B::Item: ~const Destruct,
```

Comment:
I think this highlights just how problematic the infectiousness of non-partial const trait is.

const Iterator affects the Zip constructor. Zip suddenly affects an internal zip implementation detail.
And this is a (quite radioactive) specialization of that implementation detail that nobody should have to review unless it is really necessary.